lib/metadataTypes/Query.js

```diff
@@ -116,45 +116,117 @@ class Query extends MetadataType {
             this.definition.targetUpdateTypeMapping[metadata.targetUpdateTypeName];
         return metadata;
     }
+    /**
+     * helper for buildDefinitionForExtracts
+     * searches extracted SQL file for template variables and applies the market values
+     *
+     * @param {string} code code from extracted code
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @returns {string} code with markets applied
+     */
+    static applyTemplateValues(code, templateVariables) {
+        // fix bad formatting applied by SQL Formatter Plus
+        code = code
+            .split(' { { { ')
+            .join('{{{')
+            .split('{ { { ')
+            .join('{{{')
+            .split(' } } } ')
+            .join('}}}')
+            .split(' } } }')
+            .join('}}}');
 
+        // replace template variables with their values
+        return Mustache.render(code, templateVariables);
+    }
     /**
      * helper for buildDefinition
      * handles extracted code if any are found for complex types
      *
      * @param {string} templateDir Directory where metadata templates are stored
      * @param {string|string[]} targetDir (List of) Directory where built definitions will be saved
-     * @param {TYPE.MetadataTypeItem} metadata main JSON file that was read from file system
-     * @param {object} variables variables to be replaced in the metadata
+     * @param {TYPE.QueryItem} metadata main JSON file that was read from file system
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
      * @param {string} templateName name of the template to be built
-     * @returns {Promise.<void>} Promise
+     * @returns {Promise.<void>} void
      */
-    static async buildDefinitionForExtracts(
+    static buildDefinitionForExtracts(
         templateDir,
         targetDir,
         metadata,
-        variables,
+        templateVariables,
         templateName
+    ) {
+        return this._buildXForExtracts(
+            templateDir,
+            targetDir,
+            metadata,
+            templateVariables,
+            templateName,
+            'definition'
+        );
+    }
+    /**
+     * helper for buildTemplate
+     * handles extracted code if any are found for complex types
+     *
+     * @param {string} templateDir Directory where metadata templates are stored
+     * @param {string|string[]} targetDir (List of) Directory where built definitions will be saved
+     * @param {TYPE.QueryItem} metadata main JSON file that was read from file system
+     * @param {TYPE.TemplateMap} templateVariables variables to be replaced in the metadata
+     * @param {string} templateName name of the template to be built
+     * @returns {Promise.<void>} void
+     */
+    static buildTemplateForExtracts(
```

Comment:
suggestions for new name for clarity

- buildTemplateForNested
- extractNestedForTemplate

As it stands it sounds a bit like a data extract, and having then buildXForExtracts makes it worse. 
buildXForExtracts, could be then
- buildForNested
- buildNestedMetadata
for example